srv/src/main/java/my/bookshop/handlers/CatalogServiceHandler.java

```diff
@@ -127,11 +143,20 @@ public void afterAddReview(BooksAddReviewContext context) {
 	@After(event = CqnService.EVENT_READ)
 	public void discountBooks(Stream<Books> books) {
 		books.filter(b -> b.getTitle() != null).forEach(b -> {
-			loadStockIfNotSet(b);
 			discountBooksWithMoreThan111Stock(b, featureToggles.isEnabled("discount"));
 		});
 	}
 
+	@After(event = CqnService.EVENT_READ)
+	public void bigBooks(Stream<Books> books) {
+		books.filter(b -> b.getDetails() != null).forEach(b -> {
+			int pages = (int) b.getDetails().get("pages");
```

Comment:
What if pages are not set, but `null`? This will crash the request. There is no guarantee that pages are set, especially for new books.

Reply:
I tried it with a new book and it worked, because details is already null and the filter catches this already. With the current bookshop I don't think it breaks here. Ok, we can add an additional null check for pages.

Reply:
I think I find this a strange concept here in general. Isn't the idea behind `cds.Map` that the data is generic and arbitrary and thus I can't really write business logic on it? From a modelling perspective I'd say a better practice would be to model `pages` explicitly. I get that this is a showcase of the feature, but I somehow fear that this example might lead to inappropriate usage. @agoerler what do you think?